### DISCUSSION TITLE
Is there a way of recording handled exception from the native side?

### DISCUSSION THREAD

**Discussion by powerserg17-bunch** [invertase/react-native-firebase#5010] [Q&A]
Is there a way of recording an exception (natively `FirebaseCrashlytics.getInstance().recordException(exception)`), handled in `try catch` block in native Android code? The Crashlytics instance is not exposed natively and with RN Bridge - ReactApplicationContext does not emit an event when it crashes on startup.

**Comment by mikehardy** [comment]
Good question, I'm not sure? We have a history of making native bits public though when necessary to allow things like this, we know they can be quite useful. Is there some method that could be public that would help you accomplish your goal? If yes and you PR a change to the java I could merge it in

**Reply by powerserg17-bunch** [reply]
@mikehardy, My local copy has it as following:
```
package io.invertase.firebase.crashlytics;

import com.google.firebase.crashlytics.FirebaseCrashlytics;

public class ReactNativeFirebaseCrashlyticsNativeHelper {

  public static void recordNativeException(Throwable throwable) {
    FirebaseCrashlytics.getInstance().recordException(throwable);
  }

}
```
Please let me know if that is something you are willing to accept in the repo and I will create a PR.

**Reply by mikehardy** [reply]
I think it might be easier to have a method in an existing class? Unless there is no suitable / coherent location. But in general, something like this seems quite useful, post a PR up and I'll have a serious like + likely merge (or reshape a bit, then merge)

**Reply by powerserg17-bunch** [reply]
Editing existing classes did not seem appropriate, since they do not expose any native functionality (e.g. Module contains React methods and InitProvider has public config(onCreate) method and instance flags getters). I've made the helper class, so that it can serve the purpose of accessing methods from the native Crashlytics instance, and can be modified in the future to support more methods as needed. I just feel it provides more context on what it is. I'm submitting a PR now, let's continue the conversation there. Thank you.
